### DISCUSSION TITLE
Run progress concurrently with other outputs, and pin it to bottom

### DISCUSSION THREAD

**Discussion by nbsp** [charmbracelet/bubbletea#817] [Q&A] [answered]
I'm trying to make a program that logs to stdout while the progress bar tracks the overall progress.

Currently the problem is that bubbletea isn't able to run concurrently with the logs, instead having to finish before they start.
Another thing I couldn't figure out is how to make it pin to the bottom. I saw #179, but didn't understand how to make it work for me with arbitrary output running outside of the bubbletea process.

![image](https://github.com/charmbracelet/bubbletea/assets/57151943/4f791ab8-3295-46ff-96aa-e54e3730924d)

**Comment by meowgorithm** [comment, marked as answer]
Hi! Have a look at the [package manager example](https://github.com/charmbracelet/bubbletea/tree/master/examples/package-manager) which I'm guessing will get you most of the way there.

The particular commands I believe you want are [`tea.Printf`](https://pkg.go.dev/github.com/charmbracelet/bubbletea#Printf) and [`tea.Println`](https://pkg.go.dev/github.com/charmbracelet/bubbletea#Println).

**Comment by bashbunni** [comment]
Hey @solfisher I know it's been a while since you asked about this. Let me know if anything is still unclear here. I just marked the question as answered since that should give you the solution you're looking for, but of course if you have any questions, let us know!
